stdlib/analyze.pkl

```diff
@@ -0,0 +1,44 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+/// A library for statically analyzing Pkl modules.
+///
+/// These tools differentiate from [pkl:reflect][reflect] in that they parse Pkl modules, but do not
+/// execute any code within these modules.
+@Since { version = "0.27.0" }
+@ModuleInfo { minPklVersion = "0.27.0" }
+module pkl.analyze
```

Comment:
This seems to be the first stdlib module name that's clearly a verb instead of a noun. Bringing this up because of consistency.

Reply:
We also have "reflect"

Reply:
In my mind, "reflect" is short for "reflection". What worries me is that `analyze.importGraph()` inverts the convention of using nouns for classes/modules and verbs for functions.

Reply:
I actually read "reflect" as a verb, and `reflect.Class` as an imperative (e.g. "please reflect this class") 🤷

Reply:
I read `function importGraph()` as "import the graph", but `analyze.importGraph()` as "analyze the import graph". I guess that's why this doesn't feel quite right to me.